doc/src/EntitySpecMeta.dox

```diff
@@ -15,177 +15,309 @@
  * model or a script.
  *
  * At its core OpenAssetIO has no detailed understanding of the
- * specifics of any given entity, only that conceptually they may or may
- * not exist, and may or may not have multiple versions or relationships
- * to other entities.
+ * specifics of any given entity, only that:
+ *  - Conceptually they may or may not exist.
+ *  - Entities conform to one or more @ref trait "traits" that define
+ *    what data is associated with the entity.
+ *  - They may (or may not) have multiple versions.
+ *  - They may (or may not) have relationships to other entities.
+ *
+ * An @ref entity_reference is the token used to identify a specific
+ * entity within the system. These references are (generally) in the
+ * form of a <a
+ * href="https://en.wikipedia.org/wiki/Uniform_Resource_Identifier">URI</a>. The
+ * format of the reference itself is owned by the @ref manager and is
+ * conceptually opaque to the host. Because of this (and the fact that
+ * it is specifically only an identifier), no concrete information about
+ * an entity can be determined from its reference alone.
+ *
+ * OpenAssetIO purpose is to define a standardized set of actions that
+ * can be performed to retrieve information about an entity from one of
+ * these opaque references, or to create new ones.
+ *
+ * @section Traits
+ *
+ * Along with other things in OpenAssetIO, entities are described by
+ * a collection of 'traits'. A trait represents a concrete behaviour
```

Comment:
[question] is "behaviour" the correct word here? A quick google for the definition of "trait" yields "a distinguishing quality or characteristic". So maybe "A trait represents a distinguishing characteristic of that thing"?

Reply:
A much better suggestion, applied [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/249/commits/5bafd515075bad59684f858f92a2a4b4e961c3bc).